src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
+            'TABLE_NAME'
+        );
+
+        $tables = [];
+
+        foreach (array_keys($columnsData) as $tableName) {
+            $columns = $this->_getPortableTableColumnList(
+                $tableName,
+                '',
+                $columnsData[$tableName]
+            );
+
+            $foreignKeys = [];
+            if (! empty($foreignKeysData[$tableName])) {
+                $foreignKeys = $this->_getPortableTableForeignKeysList(
+                    $foreignKeysData[$tableName]
+                );
+            }
+
+            $indexes = [];
+            if (! empty($indexesData[$tableName])) {
+                $indexes = $this->_getPortableTableIndexesList(
+                    $indexesData[$tableName],
+                    $tableName
+                );
+            }
+
+            $table = new Table($tableName, $columns, $indexes, [], $foreignKeys, []);
+            if (! empty($columnsData[$tableName][0]['TABLE_COMMENTS'])) {
+                $table->addOption('comment', $columnsData[$tableName][0]['TABLE_COMMENTS']);
+            }
+
+            $tables[] = $table;
+        }
+
+        return $tables;
+    }
+
+    /**
+     * Gets all column definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseColumns(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT C.OWNER,
+                 C.TABLE_NAME,
+                 C.COLUMN_NAME,
+                 C.DATA_TYPE,
+                 C.DATA_DEFAULT,
+                 C.DATA_PRECISION,
+                 C.DATA_SCALE,
+                 C.CHAR_LENGTH,
+                 C.DATA_LENGTH,
+                 C.NULLABLE,
+                 D.COMMENTS,
+                 E.COMMENTS TABLE_COMMENTS
+            FROM ALL_TAB_COLUMNS C
+       LEFT JOIN ALL_TAB_COMMENTS E ON E.OWNER = C.OWNER AND E.TABLE_NAME = C.TABLE_NAME
+       LEFT JOIN ALL_COL_COMMENTS D ON D.OWNER = C.OWNER AND D.TABLE_NAME = C.TABLE_NAME AND
+                 D.COLUMN_NAME = C.COLUMN_NAME
+SQL;
+
+        $conditions = [];
+        $params     = [];
+
+        $conditions[]    = 'C.OWNER = :OWNER';
+        $params['OWNER'] = $databaseName;
+
+        if ($tableName !== null) {
+            $conditions[]         = 'C.TABLE_NAME = :TABLE_NAME';
+            $params['TABLE_NAME'] = $tableName;
+        }
+
+        $sql .= ' WHERE ' . implode(' AND ', $conditions);
+        $sql .= <<<'SQL'
+        ORDER BY C.TABLE_NAME, C.COLUMN_ID
+SQL;
+
+        return $this->_conn->fetchAllAssociative($sql, $params);
+    }
+
+    /**
+     * Gets all indexes definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseIndexes(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT IND_COL.TABLE_NAME,
+                 IND_COL.INDEX_NAME AS NAME,
```

Comment:
All aliases here must be redundant.

Reply:
Uhm, these aliases are used in `_getPortableTableIndexesList()` which involves calling parent... so I prefer not to touch them.